Wallet

Mingling94 opened the pull request:
Dealt with all linting errors and so this fixes #8

Wallet also has minimal functionality. Need to implement send money still

seveibar commented:
Hmm, for consistency, could you change 

```javascript
mainWindow = new BrowserWindow({
       width: 1200,
       height: 800,
       "min-width": 800,
       "min-height": 600,
       title: "Sia"
});
```
to

```javascript
mainWindow = new BrowserWindow({
       "width": 1200,
       "height": 800,
       "min-width": 800,
       "min-height": 600,
       "title": "Sia"
});
```

Thanks!

zlalvani commented:
Done.

zlalvani commented:
Done.

seveibar commented:
Awesome thanks!

Mingling94 committed:
Wallet plugin in progress

JS functions designed, HTML and CSS yet to be styled and populated
effectively

Mingling94 committed:
Linted

Moved global variable declarations to the index.html, made various
changes, continued some developing wallet plugin.

Mingling94 committed:
Wallet plugin shows list of addresses now

Dealt with all linting errors via adding .jshintignore that ignores
index.js. It was triggering errors of redefining Path that is global to
the UI renderer process.

Mingling94 committed:
Wallet plugin basics finished

Mingling94 commented:
Fixes #23 because the issues were caused by zoom.

Mingling94 committed:
Converted base unit terminology to hastings

Mingling94 committed:
Added view and button transitions

Mingling94 committed:
Got Wallet units working

Plugin uses a dropdown menu with two metric 1000 levels above and below

Data verification was really hard to get the hang of with BigNumer.js

Also replaced most single quotes with double quotes in the project

Mingling94 commented:
Boop, should be ready for merging @DavidVorick @agundy

Mingling94 committed:
Made transition instantly responsive

It's .5 seconds, .25 in and out. Though there is a 2-3 second fadein I
thought looked cool to start the app.

lukechampine commented:
these commits are interwoven, which makes them harder to review. Indentation changes, for example, are made throughout, instead of in one single linting commit. And your "Linted" commit changes wallet code! Please be more vigilant about making your commits atomic.

lukechampine commented on app/css/general.css:
```diff
@@ -83,16 +84,18 @@ html,body {
 	height: 60px;
 
+	transition: all .25s;
+	-webkit-animation: fadein 2s;
```
I would have to see it myself to be sure, but I feel like a 2sec fade in would become annoying after starting the app a few times.

DavidVorick replied:
I don't know if I'm making this number up, but I thought 400ms was a standard upper bound on stuff like this.

lukechampine commented on app/css/general.css:
```diff
@@ -110,6 +114,14 @@ html,body {
+@keyframes fadein {
+	from { opacity: 0; }
+	to   { opacity: default; }
+}
```
why do you `fadein` to `default` opacity, but `fadeout` from `1`? Does this mean all fade out events will cause the element to jump to 100% opacity before fading?

Mingling94 replied:
Small note I realized just now on Github line-note commenting. Comment on the exact line otherwise the automatic "___ commented on an outdated diff" won't autohide it.
![selection_005](https://cloud.githubusercontent.com/assets/4606838/9038325/732e4d98-39c1-11e5-85bf-8ecff8bd0464.png)

lukechampine replied:
true. In this case I commented below to show the whole context. I guess it's a tradeoff.

lukechampine commented on app/index.js:
```diff
@@ -1,11 +1,12 @@
+const App = require("app");
+const Path = require("path");
+const BrowserWindow = require("browser-window");
+const Tray = require("tray");
```
what's the rationale for `'` -> `"`? The [Google JavaScript Style Guide](http://google-styleguide.googlecode.com/svn/trunk/javascriptguide.xml?showone=Strings#Strings) recommends `'` for consistency.

DavidVorick replied:
I will add that I typically prefer `'` to `"`

Mingling94 replied:
So I've been doing a bunch of Java programming where `'` is strictly for `char` literals and `"` is strictly for `String` literals.
In JS they're interchangeable so I thought to adhere to Java style but if Google says otherwise, I don't mind.

lukechampine commented on app/js/uiManager.js:
```diff
@@ -53,8 +48,8 @@ function UIManager() {
-	 * @todo Take better approach, resolution doesn't mean high dpi. Though
-	 * supposedly there's not a sure-fire way to find dpi on all platforms.
+	 * @todo Take better approach, resolution doesn"t mean high dpi. Though
+	 * supposedly there"s not a sure-fire way to find dpi on all platforms.
```
ಠ_ಠ

Mingling94 replied:
Ahahahahahah XD

Mingling94 committed:
CSS corrections

shorter initialization fade transition

.4 for buttons at David's recommendation and 1.4 for mainbar because it
has lag time to start that needs to be delayed for

Mingling94 committed:
Changed " back to ' per suggestion

Mingling94 committed:
Two linting fixes

Mingling94 commented:
Atomic commits, got it ^_^

How's it look now?

Mingling94 commented:
@DavidVorick @lukechampine boop

DavidVorick commented:
when did this fix the zoom issue? There isn't a commit that explains it.
